pkg/flowaggregator/flowaggregator.go

```diff
@@ -518,10 +523,20 @@ func (fa *flowAggregator) fillPodLabels(key ipfixintermediate.FlowKey, record ip
 	} else {
 		klog.Warningf("Get sourcePodLabels InfoElement failed: %v", err)
 	}
-	podLabelString = fa.fetchPodLabels(key.DestinationAddress)
+
+	destinationPodLabelString := ""
+	if destinationPodName, _, exist := record.GetInfoElementWithValue("destinationPodName"); exist {
```

Comment:
let's clean up the code to avoid duplication between source & destination Pods (using a helper function, and maybe with an early return if some field is missing, to avoid too much `if` nesting)